core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java

```diff
@@ -203,6 +203,295 @@ public void testAnalyticFunctionWithFilter() throws Exception {
+                "create table trades as " +
+                        "(" +
+                        "select" +
+                        " rnd_int(1,2,3) price," +
+                        " rnd_symbol('AA','BB','CC') symbol," +
+                        " timestamp_sequence(0, 100000000000) ts" +
+                        " from long_sequence(10)" +
+                        ") timestamp(ts) partition by day",
+                null,
+                true,
+                true,
+                false
+        );
+    }
+
+    @Test
+    public void testAnalyticRankFunctionWithPartitionBySymbolAndMultiOrderWildcardLast() throws Exception {
+        assertQuery("rank\tprice\tsymbol\tts\n" +
+                        "1\t1\tBB\t1970-01-01T00:00:00.000000Z\n" +
+                        "4\t2\tCC\t1970-01-02T03:46:40.000000Z\n" +
+                        "1\t2\tAA\t1970-01-03T07:33:20.000000Z\n" +
+                        "1\t1\tCC\t1970-01-04T11:20:00.000000Z\n" +
+                        "4\t2\tBB\t1970-01-05T15:06:40.000000Z\n" +
+                        "1\t1\tBB\t1970-01-06T18:53:20.000000Z\n" +
+                        "1\t1\tBB\t1970-01-07T22:40:00.000000Z\n" +
+                        "1\t1\tCC\t1970-01-09T02:26:40.000000Z\n" +
+                        "1\t1\tCC\t1970-01-10T06:13:20.000000Z\n" +
+                        "1\t2\tAA\t1970-01-11T10:00:00.000000Z\n",
+                "select rank() over(partition by symbol order by symbol, price), * from trades",
+                "create table trades as " +
+                        "(" +
+                        "select" +
+                        " rnd_int(1,2,3) price," +
+                        " rnd_symbol('AA','BB','CC') symbol," +
+                        " timestamp_sequence(0, 100000000000) ts" +
+                        " from long_sequence(10)" +
+                        ") timestamp(ts) partition by day",
+                null,
+                true,
+                true,
+                false
+        );
+    }
+
+    @Test
+    public void testAnalyticRankFunctionWithNoPartitionByAndOrderBySymbolWildcardLast() throws Exception {
+        assertQuery("rank\tprice\tsymbol\tts\n" +
+                        "3\t1\tBB\t1970-01-01T00:00:00.000000Z\n" +
+                        "7\t2\tCC\t1970-01-02T03:46:40.000000Z\n" +
+                        "1\t2\tAA\t1970-01-03T07:33:20.000000Z\n" +
+                        "7\t1\tCC\t1970-01-04T11:20:00.000000Z\n" +
+                        "3\t2\tBB\t1970-01-05T15:06:40.000000Z\n" +
+                        "3\t1\tBB\t1970-01-06T18:53:20.000000Z\n" +
+                        "3\t1\tBB\t1970-01-07T22:40:00.000000Z\n" +
+                        "7\t1\tCC\t1970-01-09T02:26:40.000000Z\n" +
+                        "7\t1\tCC\t1970-01-10T06:13:20.000000Z\n" +
+                        "1\t2\tAA\t1970-01-11T10:00:00.000000Z\n",
+                "select rank() over(order by symbol), * from trades",
```

Comment:
Could you also add a test for the `rank() over ()` case?

Reply:
Of course

Reply:
Maybe i need modify analytic logic in `SqlCodeGenerator`, because we can't check `over()` existed  in `RankFunctionFactory` or `RowNumberFunctionFactory`

Reply:
It's probably fine if we don't support `rank() over ()`, but we should throw a `SqlException` in this case instead of `return LongConstant.newInstance(1);` and have a test to verify that.

Reply:
support rank() over() now.